### PR TITLE
univention: add common code for univention corporate server modules

### PR DESCRIPTION
**Pull request by keachi**
##### ISSUE TYPE
- Feature Pull Request
##### ANSIBLE VERSION

```
ansible 2.2.0
```
##### SUMMARY

This is common code for univention corporate server modules:
actually there are some PR on the ansible-modules-extras repository:
- [udm_dns_zone](https://github.com/ansible/ansible-modules-extras/pull/2382)
- [udm_dns_record](https://github.com/ansible/ansible-modules-extras/pull/2394)
- [udm_group](https://github.com/ansible/ansible-modules-extras/pull/2396)
- [udm_user](https://github.com/ansible/ansible-modules-extras/pull/2406)
- [udm_share](https://github.com/ansible/ansible-modules-extras/pull/2439)

Additional modules will follow with reference to this PR.

### CODE REVIEW

File: lib/ansible/module_utils/univention.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
```
Comment (bcoca) [inline lib/ansible/module_utils/univention.py]: this file is an import, no shebang needed

File: lib/ansible/module_utils/univention.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+"""univention corporate server access module"""
+
+try:
+    import univention.uldap
+    import univention.config_registry
+    import univention.admin.uldap
+    import univention.admin.objects
+    import univention.admin.config
+    import re
```
Comment (bcoca) [inline lib/ansible/module_utils/univention.py]: not all these imports belong here, they should have specific messages if dependencies other than univention are missing

**Comment by 2-B** [comment]
We changed the whole import process, because the import should be executed on the target machine.
I removed the shebang.

**Comment by keachi** [comment]
I removed the shebang, thanks for reviewing.
The whole import should be executed on the target machine, so we changed the import process.

**Comment by resmo** [comment]
You used GPL, which is fine for module but for module_utils, a BSD style license is preferred . The reason is that closed source modules can be distributed using the code in module_utils.

**Comment by keachi** [comment]
I/we prefer GPL, because we want, that other modules for univention are also open source.

**Comment by resmo** [comment]
I understand, I am ok with it but would like to get the ok from ansible core. /cc @bcoca

**Comment by resmo** [comment]
I brought this up to the Public Core Meeting (see log https://meetbot.fedoraproject.org/ansible-meeting/2016-08-09/ansible_core_public_meeting.2016-08-09-19.07.html)

I am afraid, Ansible insists on all items below module_utils/ not being GPL.

**Comment by resmo** [comment]
As an option you could keep GPL but move your shared code to `ansible-modules-extras/univention/__init__.py` and adjust your include to it in your modules e.g. `from ansible.modules.extras.univention import ...`

**Comment by resmo** [comment]
ignore my last comment, this won't work. Never mind.

**Comment by keachi** [comment]
I got the rights to relicense to BSD-2-clause.

**Comment by resmo** [comment]
kicking the build again

**Comment by resmo** [comment]
LGTM

**Pull request merged**